sdk/keyvault/keyvault-keys/src/localCryptography/algorithms.ts

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { publicEncrypt, createVerify } from "crypto";
+import * as constants from "constants";
+import { isNode } from "@azure/core-http";
+import { JsonWebKey, KeyOperation } from "../keysModels";
+import { LocalCryptographyUnsupportedError } from "./models";
+import { createHash } from "./hash";
+
+/**
+ * This file contains the implementation of local supported algorithms, which
+ * consist of wrappers around methods already available as part of NodeJS.
+ *
+ * The goal of this file is to abstract away how we implement these algorithms,
+ * since we might be using environment-specific code or dependencies.
+ *
+ * Some algorithms can't be implemented neither in the Browser or in Node
+ * without third party dependencies or considerable development.
+ *
+ * We currently don't support any operation that requires the private data of a
+ * key. Once we implement support for the private part of JWK keys and PEM keys,
+ * we will be able to increase the support of our existing algorithms.
+ */
+
+/**
+ * @internal
+ * @ignore
+ * Abstract representation of a validator.
+ * Validators verify that the requirements to execute a local cryptography operation are met.
+ * @param key The JSON Web Key that will be used during the local operation.
+ * @param operationName The name of the operation, as in "encrypt", "decrypt", "sign", etc.
+ */
+export type LocalValidator = (
+  key?: JsonWebKey,
+  operationName?: LocalCryptographyOperationName
+) => void;
+
+/**
+ * @internal
+ * @ignore
+ * The list of known validators so far.
+ * Validators verify that the requirements to execute a local cryptography operation are met.
+ */
+export const validators: Record<"keyOps" | "rsa" | "nodeOnly", LocalValidator> = {
```

Comment:
it feels a little weird for validators to throw directly, maybe they should be called assertions instead?

Reply:
I like assertions! Thank you!

Reply:
I moved all the validators to assertions, however I kept the LocalSupportedAlgorithm method "validate", since "assert" felt wrong? Let me know if you have a suggestion for that.